File: maxhanna.client/src/services/datacontracts/nexus/unit-stats.ts
import { FileEntry } from "../file/file-entry";

export class UnitStats {
  id!: number;
  unitId!: number;
  unitType!: string;
  unitLevel!: number;
  duration!: number;
  cost!: number;
  speed!: number;
  supply!: number;
  goldCarryingCapacity!: number;
  groundDamage!: number;
  airDamage!: number;
  buildingDamage!: number;
  starportLevel!: number;
  factoryLevel!: number;
  engineeringBayLevel!: number;
  pictureSrc?: string;
  purchasedValue?: number;
  sentValue?: number;
}

File: maxhanna.Server/Controllers/DataContracts/Nexus/UnitStats.cs
namespace maxhanna.Server.Controllers.DataContracts.Nexus
{
    public class UnitStats
    {
        public int Id { get; set; }
        public int UnitId { get; set; }
        public string? UnitType { get; set; }
        public int UnitLevel { get; set; }
        public int Duration { get; set; }
        public int Cost { get; set; }
        public int Supply { get; set; }
        public Decimal Speed { get; set; } 
        public int FactoryLevel { get; set; }
        public int EngineeringBayLevel { get; set; }
        public int StarportLevel { get; set; }
        public int AirDamage { get; set; }
        public int GroundDamage { get; set; }
        public int BuildingDamage { get; set; }
        public int? SentValue { get; set; }

    }
}
